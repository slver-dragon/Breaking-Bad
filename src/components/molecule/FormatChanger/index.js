import React from "react";
import Button from "../../atom/Button";
import cardsListOn from "../../../assets/img/listOn.png";
import cardsListOff from "../../../assets/img/listOff.png";
import cardsGridOn from "../../../assets/img/gridOn.png";
import cardsGridOff from "../../../assets/img/gridOff.png";
import style from "./style.module.scss";

const FormatChanger = ({ isCardsList, exchangeCatalogFormat }) => {
  const [cardsListImage, cardsGridImage] = isCardsList
    ? [cardsListOn, cardsGridOff]
    : [cardsListOff, cardsGridOn];
  return (
    <div className={style.changer}>
      <span>Каталог</span>
      <div className={style.buttonBlock}>
        <span id="PutListFormat" onClick={() => exchangeCatalogFormat(true)}>
          <Button img={cardsListImage} text={""} />
        </span>
        <span id="PutGridFormat" onClick={() => exchangeCatalogFormat(false)}>
          <Button img={cardsGridImage} text={""} />
        </span>
      </div>
    </div>
  );
};

export default FormatChanger;
