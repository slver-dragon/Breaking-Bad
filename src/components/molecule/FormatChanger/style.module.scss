@import "../../../assets/styles/style";
.changer {
  display: inline-flex;
  justify-content: space-between;
  padding: 0.5rem 0;
  span {
    @include personPageBigFont;
  }
  .buttonBlock {
    @include buttonBlock;
  }
}
