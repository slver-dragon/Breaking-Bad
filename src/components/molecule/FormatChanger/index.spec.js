import React from "react";
import { mount, shallow } from "enzyme";
import FormatChanger from "../FormatChanger"

describe("Testing molecule FormatChanger:", () => {
    let component = null;
    const onBtnClick = jest.fn();
    const setComponent = (props) => shallow(<FormatChanger {...props} />);

    test("is rendering", () => {
      component = setComponent();
      expect(component).toMatchSnapshot()
    });
  
    test("onClick works", () => {
      component = shallow(<FormatChanger exchangeCatalogFormat={onBtnClick} />);
      component.childAt(1).childAt(0).simulate("click");
      expect(onBtnClick).toHaveBeenCalled()
      component = shallow(<FormatChanger exchangeCatalogFormat={onBtnClick} />);
      component.childAt(1).childAt(1).simulate("click");
      expect(onBtnClick).toHaveBeenCalled()
    });

    test("output mode works", () => {
      component = mount(<FormatChanger isCardList="true" />);
      console.log(component)
      // expect(component).toMatchSnapshot()
    });
  });
  