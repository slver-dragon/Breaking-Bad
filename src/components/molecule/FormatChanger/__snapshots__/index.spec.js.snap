// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing molecule FormatChanger: is rendering 1`] = `
<div
  className="changer"
>
  <span>
    Каталог
  </span>
  <div
    className="buttonBlock"
  >
    <span
      onClick={[Function]}
    >
      <Button
        img="test-file-stub"
        text=""
      />
    </span>
    <span
      onClick={[Function]}
    >
      <Button
        img="test-file-stub"
        text=""
      />
    </span>
  </div>
</div>
`;
